Add Emby

djs52 opened the pull request:
**Description**
<!--
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
-->
Add Emby Media Server, based on the Jellyfin chart, using the linuxserver.io Docker image. Fixes #64 

**Type of change**

- [X] Feature/App addition
- [ ] Bugfix
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor of current code

**How Has This Been Tested?**
<!--
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
-->
Installed and running perfectly on TrueNAS-SCALE-21.03-MASTER-20210304-232915

**Notes:**
<!-- Please enter any other relevant information here -->

**Checklist:**

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [x] I have added tests to this description that prove my fix is effective or that my feature works

djs52 committed:
Simple Emby chart based on the existing Jellyfin chart

djs52 committed:
Set current UID and GID environment variables

djs52 committed:
Reset the version to match the other charts

djs52 committed:
Clean up incorrect version

djs52 committed:
Fix versions and maintainer in Chart.yaml# Please enter the commit message for your changes. Lines starting

djs52 committed:
Switch to the linuxserver.io Docker image -- it's more closely matched to other Truecharts

Ornias1993 commented on charts/emby/1.6.3/Chart.yaml:
```diff
@@ -0,0 +1,34 @@
+apiVersion: v2
+kubeVersion: ">=1.16.0-0"
+name: emby
+version: 1.6.3
+appVersion:  4.5.4.0
+description: Emby Server is a home media server
```
try keeping the upstreamVersion here, but comment it out (see other examples without upstream)

djs52 replied:
Done

Ornias1993 commented on charts/emby/1.6.3/README.md:
```diff
@@ -0,0 +1,55 @@
+# Introduction
```
Don't worry about this file, it gets automatically replaced/removed

djs52 replied:
Should I drop it from the PR?

Ornias1993 replied:
No it needs to be there, don't worry about it was all I was saying.

Ornias1993 commented on charts/emby/1.6.3/app-readme.md:
```diff
@@ -0,0 +1,2 @@
+Emby Server is a home media server
```
This file also gets automatically removed/replaced

Ornias1993 commented on charts/emby/1.6.3/Chart.yaml:
```diff
@@ -0,0 +1,34 @@
+home: https://github.com/djs52/charts/tree/master/charts/emby
+icon_url: https://raw.githubusercontent.com/djs52/charts/master/charts/emby/icon.png?raw=true
+keywords:
+  - jellyfin
+  - plex
```
I don't think it's fair to add keywords for totally different solutions...

djs52 replied:
Heh :) I literally copied this list from the jellyfin chart! I was a bit surprised, but figured it was fair game.

Ornias1993 replied:
Yeah needs to go there too... thanks for the headsup! :)

Ornias1993 commented on charts/emby/item.yaml:
```diff
@@ -0,0 +1,3 @@
+categories:
+  - media
+icon_url: https://raw.githubusercontent.com/djs52/charts/master/charts/emby/icon.png?raw=true
```
end of file error

djs52 replied:
Done

Crow-Control reviewed (changes requested):
Looks quite good, some small nitpicks. 💯 
However, I can't accept this to master, as master does not accept new apps due to the refactor going on.
Accepting this would mean I would directly need to rewrite your questions.yaml, values.yaml and test_values.yaml and I rather focus on getting the refactors done ;-)

Please submit/pr this to "staging" instead. (as requested on the IX forums)

Besides that big thing and the nitpicks, it's quite good.
If it wheren't for the refactor it would be merged already!

Also:
1 more nit:
- if you want to be the maintainer, try adding yourself to .github/CODEOWNERS

Upside:
The documentation and testing in the staging branch is much improved for the things that are refactored :)

djs52 commented:
> Looks quite good, some small nitpicks. 
> However, I can't accept this to master, as master does not accept new apps due to the refactor going on.
> Accepting this would mean I would directly need to rewrite your questions.yaml, values.yaml and test_values.yaml and I rather focus on getting the refactors done ;-)

That's fair. Anything I need to do to prepare this for the post-refactor world?

> Please submit/pr this to "staging" instead. (as requested on the IX forums)

Hah, I hadn't even noticed Truecharts had its own thread. Sorry about that, switched the PR over.

> Besides that big thing and the nitpicks, it's quite good.

Thanks :)

djs52 committed:
Add end of file \n

djs52 committed:
Add back upstream_version (commented out)

djs52 committed:
Update CODEOWNERS

Ornias1993 commented:
I'm going to consider this blocked till #219 gets merged into staging...
As thats the second part of the two-part big breaking-change refactor.

Ornias1993 commented on charts/emby/1.6.3/Chart.yaml:
```diff
@@ -0,0 +1,35 @@
+  - https://github.com/truecharts/charts/tree/master/charts/emby
+  - https://hub.docker.com/r/linuxserver/emby
+  - https://github.com/linuxserver/docker-emby.git
+dependencies:
+  - name: common
+    repository: https://charts.truecharts.org/
+    version: 1.6.5
+    # condition:
+    # tags:
+    # import-values:
+    # alias:
```
for staging this needs to be 2.0.0
(and needs to be manually copied)

Ornias1993 commented on charts/emby/1.6.3/questions.yaml:
```diff
@@ -0,0 +1,379 @@
+
+## TrueCharts Specific
+
+  - variable: appVolumeMounts
```
Make sure these follow the standardised documented format on the staging branch (.github/docs is the root folder for the docs for now, will be moved on staging to docs/)

Ornias1993 commented on charts/emby/1.6.3/questions.yaml:
```diff
@@ -0,0 +1,379 @@
+                schema:
+                  type: string
+
+  - variable: service
```
Make sure these follow the standardised documented format on the staging branch (.github/docs is the root folder for the docs for now, will be moved on staging to docs/).

This one currently does not.

Ornias1993 commented:
I should have #219 merged in a day or two, I'll make it my prime priority to merge this as soon as possible afterwards.

Ornias1993 commented:
Ingress refactor (#219) has been rushed and merged into staging.
You should now have the right documentation and tool you need :)

banzo committed:
Fix FAQ link, add issue tracker link (#226)

The previous faq url points to a 404 page.
Also added a link to the Github issue tracker.

Ornias1993 commented:
I managed to spring #233 into it now to, all big changes have been merged and you should be fine to redo this for the staging branch @djs52  👍

djs52 commented:
Awesome, thanks! I might not be able to get back to this until Monday, sorry.

Ornias1993 commented:
@djs52 No problem at all, we all have other things to do too :)

Ornias1993 commented:
Try not to merge branches, you are really dirtying your PR...
Try rebasing...

djs52 commented:
Yeah, I've made a bit of a mess of things. Let me see if I can unpick this, or if I have to recreate the PR from scratch.

Ornias1993 commented:
@djs52 No problem, just take your time :)

djs52 committed:
Update Emby for Truecharts 2.0.0

Ornias1993 commented on charts/emby/2.0.0/Chart.lock:
```diff
@@ -0,0 +1,6 @@
+dependencies:
```
remove the lockfile, it's not compatbile (and actually not needed either atm.)

Ornias1993 commented on charts/emby/2.0.0/ix_values.yaml:
```diff
@@ -0,0 +1,15 @@
+image:
+  repository: linuxserver/emby
+  pullPolicy: IfNotPresent
+  tag: version-4.5.4.0
+
+##
```
```suggestion
  tag: version-4.5.4.0

startAsRoot: true

##
```

Ornias1993 commented on charts/emby/2.0.0/test_values.yaml:
```diff
@@ -0,0 +1,52 @@
+  pullPolicy: IfNotPresent
+  tag: version-4.5.4.0
+
+strategy:
+  type: Recreate
+
+services:
```
```suggestion
strategy:
  type: Recreate

startAsRoot: true

services:
```

Crow-Control reviewed (changes requested):
Only two little nitpicks :)

djs52 committed:
Update charts/emby/2.0.0/ix_values.yaml

Co-authored-by: Kjeld Schouten-Lebbing <kjeld@schouten-lebbing.nl>

djs52 committed:
Update charts/emby/2.0.0/test_values.yaml

Co-authored-by: Kjeld Schouten-Lebbing <kjeld@schouten-lebbing.nl>

djs52 committed:
Delete Chart.lock

djs52 commented:
Done, thanks for your time :)

Ornias1993 commented:
@all-contributors please add @djs52 for code

truecharts-admin commented:
This PR is locked to prevent necro-posting on closed PRs. Please create a issue or contact staff on discord if you want to further discuss this
